gmso/external/convert_hoomd.py

```diff
@@ -295,6 +293,39 @@ def _parse_particle_information(
             else 1 * base_units["mass"]
         )
         charges.append(site.charge if site.charge else 0 * u.elementary_charge)
+    # Check for rigid IDs
+    rigid_ids = [site.rigid_id for site in top.sites]
```

Comment:
The thought here is that mBuild assigns rigid IDs at the particle level, so if an mBuild compound has that information, then I was thinking a `site` in GMSO would also, so we read it from `from_mbuild`, and vice versa, write it out in `to_mbuild`.

For hoomd, rigid IDs are assigned at the particle level as well, where the ID points to the molecule they belong to.

I think either way is technically possible. In `from_build` we could disregard the particle rigid IDs, and instead use that information to just adjust `Molecule.isrigid` based on how someone sets up their particle rigid IDs in mBuild. I think this would be sensitive to the mBuild compound correctly having its hierarchy set up? Then, in the GSD writers, we could parse each molecule in the topology, and build up a list of ints that correctly maps sites to rigid body numbers. But, in both of these cases, it seems like this is a roundabout way to having the data structure in place to begin with. 

Maybe mBuild doesn't need to have rigid body information if it no longer has writers, and all rigid body stuff should originate in GMSO?

Reply:
> Maybe mBuild doesn't need to have rigid body information if it no longer has writers, and all rigid body stuff should originate in GMSO?

Yeah, I'd lean at some point to migrate towards that sort of behavior. That is almost a part of the forcefield, since it defines the treatments needed on the particles. 

For now, I think molecule.isrigid is good enough to process that information, since that's what @daico007 set up for the GROMACS writer for rigid water. The molecules are essentially tags that are at the site level, since each molecule is assigned to a site. The edge case of course being if only part of a molecule is "rigid". But I think that is a less common and more complicated case that should be handled differently.

Reply:
I agree..removing rigid body stuff from mBuild makes sense, and then we can implement however we need to (or already is) in GMSO.